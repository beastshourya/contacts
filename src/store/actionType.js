export const ADD_CONTACT = "Contacts";
export const DELETE_CONTACT = "Deletes";
export const UPDATE_CONTACT = "Update";
export const SEARCH_CONTACT = 'Search';
