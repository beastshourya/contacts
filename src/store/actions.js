import { ADD_CONTACT,DELETE_CONTACT,UPDATE_CONTACT,SEARCH_CONTACT} from "./actionType";

export const Contacts = (contact) => ({
  type: ADD_CONTACT,
  payload: contact,
});

export const deletes = (key) => ({
  type: DELETE_CONTACT,
  key: "",
});
export const Update = (contact) => ({
  type: UPDATE_CONTACT,
  payload: contact,
});
export const Search = (input) => ({
  type: SEARCH_CONTACT,
  payload:input
});
