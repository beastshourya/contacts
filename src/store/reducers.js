import { ADD_CONTACT,DELETE_CONTACT,UPDATE_CONTACT,SEARCH_CONTACT } from "./actionType";

const initialState = {
  myContacts: [
    {key: 1,
    name: 'rahul',
    lastName: 'kumar',
    phone_no: 9431154433,}
  ],
  filteredContacts: [],
};

export const mainReducer = (state = initialState, action) => {
  switch (action.type) {
    case ADD_CONTACT:
      return {
        ...state,
        myContacts: [...state.myContacts, action.payload],
        filteredContacts: [...state.myContacts, action.payload],
      };

    case DELETE_CONTACT:
      return {
        ...state,
        myContacts: state.myContacts.filter((item) => item.key !== action.key),
        filteredContacts: state.myContacts.filter(
          (item) => item.key !== action.key
        ),
      };
    case UPDATE_CONTACT:
      return {
        ...state,
        myContacts: state.myContacts.map((contact) =>
          contact.key == action.payload.key ? action.payload : contact
        ),

        filteredContacts: state.myContacts.map((contact) =>
          contact.key == action.payload.key ? action.payload : contact
        ),
      };
    case SEARCH_CONTACT:
      return {
        ...state,
        filteredContacts: state.myContacts.filter((item) => {
          return item.name.toLowerCase().includes(action.payload.toLowerCase());
        })
      };

    default:
      return state;
  }
};


