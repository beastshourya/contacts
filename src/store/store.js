import { createStore } from "redux";

import { mainReducer } from "./reducers";

import { AsyncStorage } from "react-native";
import { persistStore, persistReducer } from "redux-persist";

const persistConfig = {
  key: "persistedReducer",

  storage: AsyncStorage,
};

const persistedReducer = persistReducer(persistConfig, mainReducer);

const store = createStore(persistedReducer);

let persistor = persistStore(store);

export { store, persistor };
