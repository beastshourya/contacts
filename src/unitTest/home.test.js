import React from "react";
import renderer from "react-test-renderer";
import Home from "../Screens/home";
import { Provider } from "react-redux";
import { store } from "../store/store";

test("renders correctly", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Home />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
