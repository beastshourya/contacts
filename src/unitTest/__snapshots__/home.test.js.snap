// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 20,
          "marginLeft": 10,
          "marginTop": 12,
        }
      }
    >
      LIST OF CONTACTS
    </Text>
    <View
      style={
        {
          "marginLeft": "87%",
          "marginTop": 5,
          "position": "absolute",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text />
      </View>
    </View>
  </View>
  <TextInput
    onChangeText={[Function]}
    placeholder="Search Contacts"
    style={
      {
        "borderRadius": 20,
        "borderWidth": 1,
        "margin": 6,
        "padding": 12,
      }
    }
  />
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      data={[]}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View />
    </RCTScrollView>
  </View>
</View>
`;
